### PR TITLE
Fix: Improved multidex and repository declaration

### LINKED ISSUE #910: Bug: Unresolved dependencies error while building app
**Summary:**

Can't build the app, because of compiler unresolved dependencies errors, such as:

> Failed to resolve: multidex
> Failed to resolve: multidex-instrumentation

**Steps to reproduce:** 

1. Clone development branch
2. Open project in your IDE
3. Build project

**Expected behaviour:**

Complete build

**Observed behaviour:**

As I mentioned before, there are unresolved dependencies errors:

image

Running `./gradlew --debug check` logs:
```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:getDependencies'.
Could not find multidex.jar 
```
For more details, check pastebin where I've posted more details from logs
 https://pastebin.com/XiXbYFAW

**Device and Android version:**

Mobile device doesn't matter here, so I'll just say that I'm using Android Studio as an IDE

**Screenshots:**

image

### PR DESCRIPTION
**Pull request by etlhsu**
Fix #910: Unresolved dependencies error while building app

The commit fixes issues where certain references caused issues when building project

### CODE REVIEW

**Comment by etlhsu** [comment]
@therajanmaurya and @miPlodder, it seems that contributors are having the issue that my patch solves, people cannot build the project without my change. You may want to consider address my pull request.

**Comment by miPlodder** [comment]
@ethan627hsu I have not faced any issues. What was the build issue that you were facing? Create an issue for it.

**Comment by etlhsu** [comment]
@miPlodder Some contributors face an issue where when building, Gradle gets confused about the support library and multidex. I've faced these problems locally, and when using Travis CI. The changes proposed will fix these issues, as some pull requests from others try to patch the issue within their functional solution. 

*Screenshot of gradle build fail from @MigDinny in [gitter](https://gitter.im/openMF/self-service-app)
image

File: build.gradle [hunk 1]
```diff
@@ -19,8 +19,8 @@ buildscript {
 
 allprojects {
     repositories {
-        jcenter()
         google()
+        jcenter()
```
Comment (miPlodder) [inline build.gradle:23]: @ethan627hsu Why are you swapping these lines? This issue can be resolved by adding `multi-dex` dependency.

Comment (etlhsu) [reply build.gradle:23]: @miPlodder The swapping prevents an error when referencing recyclerview in gradle

Comment (miPlodder) [reply build.gradle:23]: @ethan627hsu Can you add the reference/link from where you got this solution?

Comment (etlhsu) [reply build.gradle:23]: @miPlodder A user on StackOverflow also experienced my same issue. [Here](https://stackoverflow.com/a/48243876) was his solution.

File: app/build.gradle [hunk 2]
```diff
@@ -134,5 +134,7 @@ dependencies {
     androidTestImplementation "com.android.support.test.espresso:espresso-core:$rootProject.espressoVersion"
     androidTestImplementation "com.android.support.test:runner:$rootProject.runnerVersion"
     androidTestImplementation "com.android.support.test:rules:$rootProject.rulesVersion"
+
```
Comment (miPlodder) [inline app/build.gradle]: @ethan627hsu Add this as last implementation statement, that is above testImplementation and androidTestImplementation.

Comment (etlhsu) [reply app/build.gradle]: Ok, added the change @miPlodder

**Comment by miPlodder** [comment]
@ethan627hsu Change the commit message to `Fix #issue_number: message`.

**Comment by etlhsu** [comment]
@miPlodder I added a modified commit message to [the commit](https://github.com/openMF/mifos-mobile/pull/909/commits/1e2d6dd63c231907cf89eb34412f0210418f8275).

**Comment by etlhsu** [comment]
@miPlodder Do I need to make any other additional changes?

**Pull request merged**